type-c-service/src/task.rs

```diff
@@ -17,16 +17,19 @@ pub async fn task_closure<'a, Fut: Future<Output = ()>, F: Fn(&'a Service) -> Fu
 
     let power_policy_channel = POWER_POLICY_CHANNEL.init(PubSubChannel::new());
     let power_policy_publisher = power_policy_channel.dyn_immediate_publisher();
-    // Guaranteed to not panic since we initialized the channel above
-    let power_policy_subscriber = power_policy_channel.dyn_subscriber().unwrap();
+    let power_policy_subscriber = if let Ok(subscriber) = power_policy_channel.dyn_subscriber() {
+        subscriber
```

Comment:
We were discussing whether or not there could be an API improvement to `PubSubChannel` and `Watch` that provides an infallible array of the subscribers when constructing the object such as a `PubSubChannel::new() -> (Self, [Subscribers; N])`